File: sdks/dotnet/sulfone-helium/Api/Processor/Mapper.cs
using sulfone_helium.Api.Core;
using sulfone_helium.Domain.Processor;

namespace sulfone_helium.Api.Processor;

public static class ProcessorMapper
{
    public static ProcessorInput ToDomain(this ProcessorReq req)
    {
        return new ProcessorInput(
            req.ReadDir,
            req.WriteDir,
            req.Globs.Select(x => x.ToDomain()),
            req.Config.ToDynamic()
        );
    }

    public static ProcessorRes ToRes(this ProcessorOutput res)
    {
        return new ProcessorRes { OutputDir = res.Directory };
    }
}

File: sdks/python/read_dir/processor/mapper.py
from api.core.core_mapper import CyanMapper
from api.processor.req import ProcessorReq
from api.processor.res import ProcessorRes
from domain.processor.input import ProcessorInput
from domain.processor.output import ProcessorOutput


class ProcessorMapper:
    @staticmethod
    def to_domain(req: ProcessorReq) -> ProcessorInput:
        globs = [CyanMapper.glob_req_to_domain(x) for x in req.globs]
        return ProcessorInput(
            read_directory=req.read_dir,
            write_directory=req.write_dir,
            config=req.config,
            globs=globs
        )

    @staticmethod
    def to_res(res: ProcessorOutput) -> ProcessorRes:
        return ProcessorRes(output_dir=res.directory)
